### DISCUSSION TITLE
Hide fields from interfaces

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#189] [Q&A] [answered]
Hi, @wu-clan 
```
class UserBase(BaseModel):
    username: str
    password: str

class CreateUser(UserBase):
    dept_id: int
```

For example, when creating a user, the department id is hidden from the interface, it is not open to the outside world, and it is set by the backend

**Comment by wu-clan** [comment]
This doesn't seem very realistic, dept_id will be fixed default value

This encompasses different scenarios, just set dept_id to an optional field, for front-end registrations, don't transmit it, for back-end registrations, transmit it

**Comment by downdawn** [comment]
This is just an example. 
Alternatively, for a particular scenario, the user ID is required, but it does not need to be passed from the frontend. However, the schema requires the user ID to be set. If it's optional, it could mislead the frontend.

**Comment by davidjb99** [comment, marked as answer]
Define a base schema and hide any specified fields, as discussed https://github.com/tiangolo/fastapi/discussions/7585

```
class MyBase(BaseModel):
    class Config:
        @staticmethod
        def schema_extra(schema: dict, _):
            props = {}
            """
            Removes any fields from the OpenAPI spec if they have a `hidden` tag
            """
            for k, v in schema.get("properties", {}).items():
                if not v.get("hidden", False):
                    props[k] = v
            schema["properties"] = props

class UserCreate(MyBase):
    id: int | None = Field(hidden=True)
   name: str

```

**Reply by wu-clan** [reply]
Thanks, this is very helpful for In

**Reply by downdawn** [reply]
Nice, we can add it to SchemaBase.

I don't know if it is compatible with V2

**Reply by wu-clan** [reply]
This feature is implemented in V2: https://docs.pydantic.dev/latest/usage/models/#extra-fields
